tests/integration/test_integration.py

```diff
@@ -751,67 +731,97 @@ async def test_given_vault_integrated_with_s3_when_create_backup_then_action_fai
         status="active",
         timeout=1000,
     )
-    await ops_test.model.integrate(
-        relation1=APP_NAME,
-        relation2=S3_INTEGRATOR_APPLICATION_NAME,
-    )
-    await ops_test.model.wait_for_idle(
-        apps=[APP_NAME],
-        status="active",
-        timeout=1000,
-        wait_for_exact_units=NUM_VAULT_UNITS,
-    )
-    vault = ops_test.model.applications[APP_NAME]
-    assert isinstance(vault, Application)
+    vault_app = get_app(ops_test.model)
+    if not has_relation(vault_app, "s3-parameters"):
+        await ops_test.model.integrate(
+            relation1=APP_NAME,
+            relation2=S3_INTEGRATOR_APPLICATION_NAME,
+        )
+        await ops_test.model.wait_for_idle(
+            apps=[APP_NAME],
+            status="active",
+            timeout=1000,
+            wait_for_exact_units=NUM_VAULT_UNITS,
+        )
     create_backup_action_output = await run_create_backup_action(ops_test)
     assert create_backup_action_output.get("return-code") == 0
 
 
 @pytest.mark.abort_on_fail
 async def test_given_vault_integrated_with_s3_when_list_backups_then_action_fails(
-    ops_test: OpsTest, deploy_requiring_charms: None
+    ops_test: OpsTest, vault_authorized: Task, s3_integrator_idle: Task
 ):
+    # TODO: Shouldn't this be then_action_succeeds?
```

Comment:
Hum, yes?

Reply:
Thanks for the sanity check. I forgot I left these in here, but that's why I do it :)

Reply:
https://github.com/canonical/vault-operator/pull/271/commits/e107919f0747991d12525151687e0f38ebd03c99